packages/lib/server/repository/organization.ts

```diff
@@ -0,0 +1,68 @@
+import { IS_TEAM_BILLING_ENABLED } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+import { safeStringify } from "@calcom/lib/safeStringify";
+import { prisma } from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import { createAProfileForAnExistingUser } from "../../createAProfileForAnExistingUser";
+
+export class OrganizationRepository {
+  static async createWithOwner({
+    orgData,
+    owner,
+  }: {
+    orgData: {
+      name: string;
+      slug: string;
+      isOrganizationConfigured: boolean;
+      autoAcceptEmail: string;
+      seats: number | null;
+      pricePerSeat: number | null;
+    };
+    owner: {
+      id: number;
+      email: string;
+      nonOrgUsername: string;
+    };
+  }) {
+    logger.debug("Creating organization with owner", safeStringify({ orgData, owner }));
+    const organization = await prisma.team.create({
+      data: {
+        name: orgData.name,
+        isOrganization: true,
+        ...(!IS_TEAM_BILLING_ENABLED ? { slug: orgData.slug } : {}),
+        organizationSettings: {
+          create: {
+            isOrganizationVerified: true,
```

Comment:
Note that we are making a verified organization in self-serve as well because there is a big paywall for this in self-serve.

Reply:
Don't see an issue with this - i guess they still need to confirm the DNS settings as they did previouly?

Reply:
DNS is automatically setup on cal.com. But he can't use the organization(it stays unpublished) till it is paid for.